server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -256,15 +256,24 @@ public boolean isLeader()
    * @return tier -> { dataSource -> underReplicationCount } map
    */
   public Map<String, Object2LongMap<String>> computeUnderReplicationCountsPerDataSourcePerTier()
+  {
+    final Iterable<DataSegment> dataSegments = segmentsMetadataManager.iterateAllUsedSegments();
+    return computeUnderReplicationCountsPerDataSourcePerTierForSegments(dataSegments);
+  }
+
+  /**
+   * @return tier -> { dataSource -> underReplicationCount } map
+   */
+  public Map<String, Object2LongMap<String>> computeUnderReplicationCountsPerDataSourcePerTierForSegments(
```

Comment:
I think it's ok that `segmentReplicantLookup` could potentially be stale since it is only updated on coordinator runs, since the refreshed segment metadata would at least ensure that the stale data would be under counting replication levels, rather than potentially falsely reporting that everything is available (when forcing refresh).

Reply:
Keeping the `segmentReplicantLookup` as possibly stale for now. This still ensures that we will never return true (available) when it’s false (not available) since we force refresh metadata. Although we can return false (not available) when it’s true (available) for up to a period of coordinator run longer for the full format response. This problem will only affects the full format. We can loop back to this if we find that having the option to force refresh the  `segmentReplicantLookup` is useful. If that is the case then we can use the existing query param, forceMetadataRefresh, to force refresh `segmentReplicantLookup` too.